build.xml

```diff
@@ -192,13 +213,97 @@
       deployRoot="${source} sprint ${version}"
     />
   </target>
+  
+  <target name="local2qa-specified-tests">
+      <!-- backup qa first -->
+      <antcall target="backup-qa">
+        <param name="version" value="${version}" />
+        <param name="type" value="package" />
+      </antcall>
+      <antcall target="backup-qa">
+        <param name="version" value="${version}" />
+        <param name="type" value="destructiveChanges" />
+      </antcall>
+      <antcall target="copy-destructiveChanges">
+        <param name="version" value="${version}" />
+        <param name="source" value="${source}" />
+      </antcall>
+  	<!-- deploy script -->
+  	<script language="javascript">

```

Comment:
Is there anyway to modularize either the JavaScript code so we do not need to cut & paste it multiple times with minor differences per environment?  Or better yet, can we use JavaScript to simply aid in creating multiple <runTest> rows?  If not, did you weigh the pros/cons of passing the tests in a an array rather than adding one by one via the for loop? 

String[] tests = {"TestClass1", "TestClass2", "TestClass3"};
deployOptions.setRunTests(tests);